docs/designs/0001-golang-1-21.md

```diff
@@ -0,0 +1,206 @@
+# Go 1.21 toolchains
+
+## Overview
+
+Go 1.21 introduced a new feature called _toolchains_ which allow usage of different Go toolchains to be used for different modules simultaneously instead of the default bundled Go toolchain (i.e. the bundled `go` binary). The toolchain feature is enabled by the `toolchain` keyword which can either be in the `go.mod` file or the `go.work` (workspaces) file [[1]](#references).
+
+This document describes how this feature works and how it impacts the current implementation in Cachi2, and proposes a few ways the implementation can be approached.
+
+## Context
+
+[What was the context for this doc? I can't think of anything important to mention here, and that's why I marked this section as optional in the template].
+
+## How it works
```

Comment:
This is too free-form. I believe we need to standardize on this H2 heading. Also, I'm still fairly convinced that the paragraphs that follow this heading are **the overview** of the 1.21 changes that are relevant to the cause, it doesn't say anything about how we're going to solve it - that's good, because that will come later in `Proposed solutions`!